econml/iv/dr/_dr.py

```diff
@@ -73,83 +246,1246 @@ def _effect_estimate(self, nuisances):
+        cross-fitting.
+
+    random_state: int, :class:`~numpy.random.mtrand.RandomState` instance or None, optional (default=None)
+        If int, random_state is the seed used by the random number generator;
+        If :class:`~numpy.random.mtrand.RandomState` instance, random_state is the random number generator;
+        If None, the random number generator is the :class:`~numpy.random.mtrand.RandomState` instance used
+        by :mod:`np.random<numpy.random>`.
+    """
+
+    def __init__(self, *,
+                 model_y_xw="auto",
+                 model_t_xw="auto",
+                 model_z_xw="auto",
+                 model_t_xwz="auto",
+                 model_tz_xw="auto",
+                 prel_model_effect,
+                 projection=False,
+                 featurizer=None,
+                 fit_cate_intercept=True,
+                 alpha='auto',
+                 n_alphas=100,
+                 alpha_cov='auto',
+                 n_alphas_cov=10,
+                 max_iter=1000,
+                 tol=1e-4,
+                 n_jobs=None,
+                 cov_clip=0.1,
+                 opt_reweighted=False,
+                 discrete_instrument=False,
+                 discrete_treatment=False,
+                 categories='auto',
+                 cv=2,
+                 mc_iters=None,
+                 mc_agg='mean',
+                 random_state=None):
+        self.alpha = alpha
+        self.n_alphas = n_alphas
+        self.alpha_cov = alpha_cov
+        self.n_alphas_cov = n_alphas_cov
+        self.max_iter = max_iter
+        self.tol = tol
+        self.n_jobs = n_jobs
+        super().__init__(model_y_xw=model_y_xw,
+                         model_t_xw=model_t_xw,
+                         model_z_xw=model_z_xw,
+                         model_t_xwz=model_t_xwz,
+                         model_tz_xw=model_tz_xw,
+                         prel_model_effect=prel_model_effect,
+                         model_final=None,
+                         projection=projection,
+                         featurizer=featurizer,
+                         fit_cate_intercept=fit_cate_intercept,
+                         cov_clip=cov_clip,
+                         opt_reweighted=opt_reweighted,
+                         discrete_instrument=discrete_instrument,
+                         discrete_treatment=discrete_treatment,
+                         categories=categories,
+                         cv=cv,
+                         mc_iters=mc_iters,
+                         mc_agg=mc_agg,
+                         random_state=random_state)
+
+    def _gen_model_final(self):
+        return DebiasedLasso(alpha=self.alpha,
+                             n_alphas=self.n_alphas,
+                             alpha_cov=self.alpha_cov,
+                             n_alphas_cov=self.n_alphas_cov,
+                             fit_intercept=False,
```

Comment:
If we switch the design then this should be:
fit_intercept = self.fit_cate_intercept